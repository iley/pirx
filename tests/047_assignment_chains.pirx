extern func main(): int {
  var a: int;
  var b: int;
  var c: int;
  var d: int;
  
  printf("Testing assignment chains:\n");
  
  // Test simple assignment chain
  a = b = c = 42;
  printf("a=%d, b=%d, c=%d\n", a, b, c);
  
  // Test longer assignment chain
  a = b = c = d = 100;
  printf("a=%d, b=%d, c=%d, d=%d\n", a, b, c, d);
  
  // Test assignment chain with expression
  a = b = c = 10 + 5;
  printf("a=%d, b=%d, c=%d\n", a, b, c);
  
  // Test assignment chain with function call
  a = b = c = factorial(3);
  printf("a=%d, b=%d, c=%d\n", a, b, c);
  
  return 0;
}

func factorial(n: int): int {
  if n <= 1 {
    return 1;
  }
  return n * factorial(n - 1);
}