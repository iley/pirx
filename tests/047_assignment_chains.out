Testing assignment chains:
a=42, b=42, c=42
a=100, b=100, c=100, d=100
a=15, b=15, c=15
a=6, b=6, c=6
